test/js/bun/http/bun-serve-html.test.ts

```diff
@@ -241,6 +241,296 @@ button:hover {
+</head>
+<body>
+  <div class="text-file-content"></div>
+</body>
+</html>
+`,
+      "styles.css": /* css */ `
+.text-file-content {
+  content: url("./example.txt");
+  display: block;
+  white-space: pre;
+  font-family: monospace;
+}
+`,
+      "example.txt": "LMAO",
+      "plugin.ts": /* ts */ `
+import type { BunPlugin } from "bun";
+
+const p: BunPlugin = {
+  name: "my-plugin",
+  setup(build) {
+    build.onLoad({ filter: /\\.txt$/ }, async ({ path }) => {
+      const text = await Bun.file(path).text();
+      return {
+        loader: "text",
+        contents: text + " OOGA BOOGA",
+      };
+    });
+  },
+};
+
+export default p;
+`,
+    });
+
+    const { subprocess, port, hostname } = await waitForServer(dir, {
+      "/": join(dir, "index.html"),
+    });
+    const response = await fetch(`http://${hostname}:${port}/`);
+    expect(response.status).toBe(200);
+    expect(response.headers.get("content-type")).toBe("text/html;charset=utf-8");
+
+    const html = await response.text();
+    expect(html).toMatchInlineSnapshot(`
+"
+<!DOCTYPE html>
+<html>
+<head>
+  
+<link rel="stylesheet" crossorigin href="/chunk-efbnbska.css"><script type="module" crossorigin src="/chunk-kdrjysff.js"></script></head>
+<body>
+  <div class="text-file-content"></div>
+</body>
+</html>
+"
+`);
+
+    const cssResponse = await fetch(`http://${hostname}:${port}/chunk-efbnbska.css`);
+    expect(cssResponse.status).toBe(200);
+    const css = await cssResponse.text();
+    // the base64 encoding of "LMAO OOGA BOOGA"
+    expect(css).toMatchInlineSnapshot(`
+"/* styles.css */
+.text-file-content {
+  content: url("data:text/plain;base64,TE1BTyBPT0dBIEJPT0dB");
+  display: block;
+  white-space: pre;
+  font-family: monospace;
+}
+"
+`);
+  });
+
+  test("serve html with failing plugin", async () => {
+    const dir = tempDirWithFiles("html-css-js-failing-plugin", {
+      "index.html": /*html*/ `
+      <!DOCTYPE html>
+      <html>
+        <head>
+          <title>Bun HTML Import Test</title>
+          <link rel="stylesheet" href="styles.css">
+          <script type="module" src="script.js"></script>
+        </head>
+        <body>
+          <div class="container">
+            <h1>Hello from Bun!</h1>
+            <button id="counter">Click me: 0</button>
+          </div>
+        </body>
+      </html>
+    `,
+      "styles.css": /*css*/ `
+      .container {
+        max-width: 800px;
+        margin: 2rem auto;
+        text-align: center;
+      }
+    `,
+      "plugin.ts": /*ts*/ `
+const p = {
+  name: "failing-plugin",
+  setup(build) {
+    build.onLoad({ filter: /\.css$/ }, async () => {
+      throw new Error("Plugin failed intentionally");
+    });
+  },
+};
+
+export default p;
+`,
+    });
+
+    const { subprocess, port, hostname } = await waitForServer(dir, {
+      "/": join(dir, "index.html"),
+    });
+    const response = await fetch(`http://${hostname}:${port}/`);
+    expect(response.status).toBe(500);
```

Comment:
Lets add a test that verifies it still 500s without crashing when failing on another request after.
```ts
{
    const response = await fetch(`http://${hostname}:${port}/`);
    expect(response.status).toBe(500);
}
```